src/Data/Str.php

```diff
@@ -103,43 +103,27 @@ public static function count(string $str, $encoding = null)
     /**
      * Check if string has atleast one uppercase.
      *
-     * @param string $str String to be checked.
+     * @param string      $str      String to be checked.
+     * @param string|null $encoding Valid encoding.
      *
      * @return bool
      */
-    public static function hasUpperCase(string $str)
+    public static function hasUpperCase(string $str, string $encoding = null): bool
```

Comment:
How about using following code?

```php
public static function hasUpperCase(string $str, ?string $encoding = null): bool
```

Reply:
would have the exact same call-signature / implication?

when providing the default ```$encoding = null``` we don't need the ? to mark is as string|null

but I could change it to be in line with count() etc where there's no typehint string and only the default value if you'd prefer that.
```php
public static function reverse(string $str, $encoding = null) :string
public static function count(string $str, $encoding = null)
```

or add typehints to all the functions so they're inline with each other

Reply:
Hi, thanks for your explanation.

I don't prefer any of this type hints and @Lablnet may decide to use.

I care about that is about consistency.

Once we decide to use one of type hint style. It should be used on whole repository.

Reply:
I think we can continue as we are going with default type hinting..

Reply:
With 7.4 and typed properties around the corner I kinda like transitioning to everything typed but nontheless, your choice. If you want them gone, it's just one commit from me away :-)

Reply:
No thanks for your time am going to Merger this PR, but lets first approve by @peter279k